src/jobs/ega/axios/egaClient.ts

```diff
@@ -53,85 +49,31 @@ import {
   Result,
   RevokePermissionsFailure,
   success,
-} from './types/results';
-import { safeParseArray, ZodResultAccumulator } from './types/zodSafeParseArray';
-import { ApprovedUser, getErrorMessage } from './utils';
+} from '../types/results';
+import { safeParseArray, ZodResultAccumulator } from '../types/zodSafeParseArray';
+import { ApprovedUser, getErrorMessage } from '../utils';
+import { fetchAccessToken, isTokenExpired } from './idpClient';
 
 const { DACS, DATASETS, PERMISSIONS, REQUESTS, USERS } = EGA_API;
 
-// initialize IDP client
-const initIdpClient = () => {
-  const {
-    ega: { authHost },
-  } = getAppConfig();
-  return axios.create({
-    baseURL: authHost,
-  });
-};
-const idpClient = initIdpClient();
+const CLIENT_NAME = 'EGA_API_CLIENT';
 
 // initialize API client
 const initApiAxiosClient = () => {
   const {
     ega: { apiUrl },
   } = getAppConfig();
-  return axios.create({
+  const client = axios.create({
     baseURL: apiUrl,
     headers: {
       'Content-Type': 'application/json',
     },
   });
+  axiosRetry(client, { retries: DEFAULT_RETRIES });
```

Comment:
Added `axios-retry` and set max retries to 3. When i ran this locally i didn't encounter any retryable errors beyond this number, but the job report on completion didn't indicate that any user/dataset request was missed. All the totals lined up to what I expected.

Reply:
Did you run or write a test to see how this behaves when it does fail 3+ times?